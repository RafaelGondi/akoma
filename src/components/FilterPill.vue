<template>
	<div>
		<span class="filter-pill_container">
			<span>{{ this.label }}: </span>
			<span>{{ this.content }} </span>
		</span>
	</div>
</template>

<script>
export default {
	props: {
		/**
		 * Conteúdo do filter pill.
		 */
		content: {
			type: String,
			default: 'Conteúdo',
			required: true,
		},
		/**
		 * Label exibida ao lado do conteúdo
		 */
		label: {
			type: String,
			default: 'Label',
			required: true,
		},
	},
	computed: {
		filterPillText(){
			return `${this.label}: ${this.content}`;
		}
	},

};
</script>
<style lang="scss" scoped>
@import '../assets/sass/app.scss';

.filter-pill_container {
	padding: 8px 16px;
	border: 1px solid $cinza-4;
	border-radius: 8px;
}
</style>
