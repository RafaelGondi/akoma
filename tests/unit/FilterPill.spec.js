import { mount, createLocalVue } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import FilterPill from '../../src/components/FilterPill.vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

test('Component is mounted properly', () => {
	const wrapper = mount(FilterPill, {
		localVue,
		propsData: {
			label: 'Idade',
			content: '0 a 120 anos',
		}
	});

	expect(wrapper).toMatchSnapshot();
});

describe("Click tests", () => {
	test("if the click event is emitted properly when the filterPill is clicked'", () => {
		const wrapper = mount(FilterPill, {
			localVue,
			propsData: {
				label: 'Idade',
				content: '0 a 120 anos',
			}
		});

		expect(wrapper.find('.filter-pill__container').exists()).toBe(true);
		wrapper.find('.filter-pill__container').trigger('click');

		expect(wrapper.emitted().click).toBeTruthy();
		expect(wrapper.emitted().click).toEqual([[true]]);
	});

	test("if the click event is not emitted when the filterPill is clicked and is desabled'", () => {
		const wrapper = mount(FilterPill, {
			localVue,
			propsData: {
				label: 'Idade',
				content: '0 a 120 anos',
				disabled: true,
			}
		});

		expect(wrapper.find('.filter-pill__container').exists()).toBe(true);
		wrapper.find('.filter-pill__container').trigger('click');

		expect(wrapper.emitted().click).toBeFalsy();
	});
});
