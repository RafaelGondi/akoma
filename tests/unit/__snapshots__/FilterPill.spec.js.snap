// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is mounted properly 1`] = `
<div><span class="filter-pill__container"><span class="filter-pill__label">Idade: </span> <span class="filter-pill__content">0 a 120 anos </span> <svg xmlns="http://www.w3.org/2000/svg" width="1.2em" height="1.2em" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-down filter-pill__chevron--down">
    <polyline points="6 9 12 15 18 9"></polyline>
  </svg></span></div>
`;
